Stub out basic docs structure

tehsmyers opened the pull request:
This lays down a basic docs structure as outlined in
https://github.com/rhpit/manageiq-cli/issues/30. I've made some
arbitrary choices that are up for discussion here, including:

- Using sphinx-apidoc to delegate api documentation to sphinx's own
  :auto*: directives
- Avoiding documenting the actual CLI commands, because that looks a
  little difficult to do in a way that's easy to maintain -- I punted
  that to basically say "run miqcli --help".
- Using pbr.packaging's "get_version" function to define the VERSION
  constant, as well as set the version in the docs config
- Documenting the constants in general with the "#:" syntax
- Other stuff caught in review that I'm forgetting :)

Closes #30

ryankwilliams commented on doc/requirements.txt:
```diff
@@ -0,0 +1 @@
+sphinx
```
We should not need to define sphinx within the requirements.txt. As far as I know any package defined within requirements.txt means they are needed to run the application. Sphinx is just needed for development when we need to build & release.

The sphinx package is already included in the test-requirements.txt. Which contains other packages for development.

```
coverage!=4.4,>=4.0
mock>=2.0.0
sphinx>=1.6.2
testtools>=2.2.0
flake8>=3.5.0
```
What do you think @seandst?

tehsmyers replied:
Those are specifically docs requirements, which I added before creating a docs dependency on miqcli itself. At that point they became redundant, so the file should just go away. +1

tehsmyers commented on miqcli/constants.py:
```diff
@@ -13,22 +13,46 @@
+
+
+#: Installed version of miqcli
+VERSION = get_version('miqcli')
```
I just noticed that if we do this, we should swap VERSION and PACKAGE, so VERSION is `get_version(PACKAGE)`

Commit:
Stub out basic docs structure

This lays down a basic docs structure as outlined in
https://github.com/rhpit/manageiq-cli/issues/30. I've made some
arbitrary choices that are up for discussion here, including:

- Using sphinx-apidoc to delegate api documentation to sphinx's own
  :auto*: directives
- Avoiding documenting the actual CLI commands, because that looks a
  little difficult to do in a way that's easy to maintain -- I punted
  that to basically say "run miqcli --help".
- Using pbr.packaging's "get_version" function to define the VERSION
  constant, as well as set the version in the docs config
- Documenting the constants in general with the "#:" syntax
- Other stuff caught in review that I'm forgetting :)

Closes #30

tvieira reviewed (approved):
LGTM

ryankwilliams commented:
LGTM

tvieira commented:
Hey @seandst , I think one of the latest commits changed the `constants.py`, very small change. Feel free to rebase with that and do a "squash and merge".
